### PR TITLE
Lime and SHAP Explainer tests

### PR DESCRIPTION
**Pull request by nehachopra27**
This PR tests https://issues.redhat.com/browse/RHOAIENG-9628 in ODH smoke and sanity at the moment.

### CODE REVIEW

File: ods_ci/tests/Resources/CLI/TrustyAI/trustyai_service.resource [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    [Documentation]    Verify the deployed model is registered with TrustyAI Service by verifying MM_PAYLOAD_PROCESSORS
+    ...                value is the TrustyAI service route
+    [Arguments]        ${namespace}=${PRJ_TITLE}
+    ${MM_PAYLOAD_PROCESSORS_Expected}=    Set Variable    http://trustyai-service.${namespace}.svc.cluster.local/consumer/kserve/v2
+    Wait Until Keyword Succeeds  3 min  20 sec  Verify Model Serving Pod Exists   ${namespace}    label_selector=modelmesh-service=modelmesh-serving
+    ${podname}=    Get Pod Name   ${namespace}    label_selector=modelmesh-service=modelmesh-serving
+    Log    Serving Runtime Podname: ${podname}
+    ${MM_PAYLOAD_PROCESSORS_Actual}=  Run  oc get pod ${podname} -n ${namespace} -o json | jq '.spec.containers[0].env[] | select(.name=="MM_PAYLOAD_PROCESSORS") | .value'
+    ${MM_PAYLOAD_PROCESSORS_Actual}=    Strip String    ${MM_PAYLOAD_PROCESSORS_Actual}    characters="
+    Should Be Equal  ${MM_PAYLOAD_PROCESSORS_Actual.strip()}  ${MM_PAYLOAD_PROCESSORS_Expected.strip()}
```
Comment (adolfo-ab) [inline ods_ci/tests/Resources/CLI/TrustyAI/trustyai_service.resource:31]: Maybe we should also check if the pods are running e.g. `oc wait --for=condition=ready pod/<pod-name> --timeout=<duration>` or similar

Comment (nehachopra27) [reply ods_ci/tests/Resources/CLI/TrustyAI/trustyai_service.resource:40]: Done in line 36 now.

File: ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    ...    oc label namespace ${namespace} modelmesh-enabled=true --overwrite
+    Should Be Equal As Integers    ${rc}    ${0}
+
+Send Model Inference Request
+    [Documentation]    Send Inference to the model
+    [Arguments]    ${namespace}   ${model_name}  ${inference_input}   ${token}
+    ${url}=    Get Model Route Via CLI    ${model_name}    ${namespace}
+    ${token}=   Generate Thanos Token
+    ${curl_cmd}=    Set Variable    curl -sk ${url} -d @${inference_input}
+    ${curl_cmd}=    Catenate    ${curl_cmd}    -H "Authorization: Bearer ${token}"
```
[the remark this answers is not in the document]
Comment (nehachopra27) [reply ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot:108]: For better readability and any easy future changes, this is the standard we have been following in ods-ci.

File: ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    ${rc}  ${output}=     Run And Return Rc And Output    ${curl_cmd}
+    Should Contain    ${output}    ${model_name}
+    Should Be Equal As Integers    ${rc}    ${0}    ${output}
+
+Get Latest Inference Id
+    [Documentation]    Get all stored inference ids through TrustyAI Service endpoints
+    [Arguments]    ${model_name}   ${trusty_route}  ${token}
+    ${url}=   Set Variable     ${trusty_route}/info/inference/ids/${model_name}?type=organic
+    ${curl_cmd}=     Set Variable    curl -sk "${url}"
```
[the remark this answers is not in the document]
Comment (nehachopra27) [reply ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot:117]: For consistency across ods-ci, I would prefer this as a standard.

File: ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    ${curl_cmd}=     Catenate    ${curl_cmd}    -H "Authorization: Bearer ${token}"
```
[the remark this answers is not in the document]
Comment (nehachopra27) [reply ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot:118]: Same as above

File: ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        ${rc}    ${out}=    Run And Return Rc And Output
+    ...    oc label namespace ${namespace} modelmesh-enabled=true --overwrite
+    Should Be Equal As Integers    ${rc}    ${0}
+
+Send Model Inference Request
+    [Documentation]    Send Inference to the model
+    [Arguments]    ${namespace}   ${model_name}  ${inference_input}   ${token}
+    ${url}=    Get Model Route Via CLI    ${model_name}    ${namespace}
+    ${token}=   Generate Thanos Token
+    ${curl_cmd}=    Set Variable    curl -sk ${url} -d @${inference_input}
```
[the remark this answers is not in the document]
Comment (nehachopra27) [reply ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot:107]: Same as above

File: ods_ci/tests/Resources/CLI/TrustyAI/trustyai_service.resource [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+${TRUSTYAI_RESOURCEPATH}=            tests/Resources/Files/TrustyAI
+${TRUSTYAI_CR_FILEPATH}=             ${TRUSTYAI_RESOURCEPATH}/trustyai_cr.yaml
+
+
+*** Keywords ***
+Install And Verify TrustyAI Service
+    [Documentation]    Install TrustyAI service CRD and verify that TrustyAI resources have spun up
+    [Arguments]        ${namespace}
+    ${return_code}    ${output}=    Run And Return Rc And Output    oc apply -f ${TRUSTYAI_CR_FILEPATH} -n ${namespace}
+    Sleep    60s    msg=Wait for Trusty Route to be created
```
[the remark this answers is not in the document]
Comment (nehachopra27) [reply ods_ci/tests/Resources/CLI/TrustyAI/trustyai_service.resource]: Done now, by using 'Wait For Pods To Be Ready'

File: ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+*** Test Cases ***
+Verify LIME/SHAP Explainers Are Availble For A Model Deployed Via CLI
+    [Documentation]    Verifies that the Lime Explainers are available on sending a request
+    [Tags]    OpenDataHub     Smoke    RHOAIENG-9628     ExcludeOnRHOAI
+    Set Project And Serving Runtime    namespace=${TEST_NS}    runtime_path=${EXPLAINERS_PATH}/odh-mlserver-1.x.yaml
+    Oc Apply    kind=Secret    src=${EXPLAINERS_PATH}/storage-config.yaml     namespace=${TEST_NS}
+    Deploy Model Via CLI    isvc_filepath=${EXPLAINERS_PATH}/housing.yaml
+    ...    namespace=${TEST_NS}
+    ${TRUSTY_ROUTE}=    Install And Verify TrustyAI Service     ${TEST_NS}
+    Wait Until Keyword Succeeds  2 min  10 sec  Verify Model Is Registered With TrustyAI Service     namespace=${TEST_NS}
```
[the remark this answers is not in the document]
Comment (nehachopra27) [reply ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot]: Increased the outer wait to 3 min 30 sec

File: ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+${MODEL_PATH_ALPHA}=                 trusty/loan_model_alpha.onnx
+${AWS_BUCKET}=                       ${S3.BUCKET_1.NAME}
+${PRJ_DESCRIPTION1}=                 Explainer Project Description 1
+${RUNTIME_NAME}=                     Explainer Test
+${FRAMEWORK_ONNX}=                   onnx - 1
+
+
+*** Test Cases ***
+Verify LIME/SHAP Explainers Are Availble For A Model Deployed Via CLI
+    [Documentation]    Verifies that the Lime Explainers are available on sending a request
```
[the remark this answers is not in the document]
Comment (nehachopra27) [reply ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot]: Done

File: ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    [Documentation]    Get all stored inference ids through TrustyAI Service endpoints
+    [Arguments]    ${model_name}   ${trusty_route}  ${token}
+    ${url}=   Set Variable     ${trusty_route}/info/inference/ids/${model_name}?type=organic
+    ${curl_cmd}=     Set Variable    curl -sk "${url}"
+    ${curl_cmd}=     Catenate    ${curl_cmd}    -H "Authorization: Bearer ${token}"
+    ${rc}  ${output}=     Run And Return Rc And Output    ${curl_cmd} | jq -r '.[-1].id'
+    Should Be Equal As Integers    ${rc}    ${0}    ${output}
+    RETURN    ${output}
+
+Request Explainers
```
[the remark this answers is not in the document]
Comment (nehachopra27) [reply ods_ci/tests/Tests/1200__ai_explainability/1202__saliency_explainers.robot]: Done

**Review by adolfo-ab — APPROVED** [review]

**Review by tonyxrmdavidson — APPROVED** [review]

**Pull request merged**